### PR TITLE
Add format enum for easy use and choose format.

### PR DESCRIPTION
**Pull request by talregev**

### CODE REVIEW

File: cv_bridge/include/cv_bridge/cv_bridge.h [hunk 1]
```diff
@@ -55,6 +55,16 @@ class CvImage;
...
+//from: http://docs.opencv.org/modules/highgui/doc/reading_and_writing_images_and_video.html#Mat imread(const string& filename, int flags)
+typedef enum {
+	BMP, DIP,
+	JPG, JPEG, JPE,
+	JP2,
+	PNG,
+	PBM, PGM, PPM,
+	SR, RAS,
+	TIFF, TIF,
+} Foramt;
```
Comment (vrabaud) [inline cv_bridge/include/cv_bridge/cv_bridge.h]: please name it `Format`

Comment (talregev) [inline cv_bridge/include/cv_bridge/cv_bridge.h]: It already have name "Format"

Comment (vrabaud) [inline cv_bridge/include/cv_bridge/cv_bridge.h]: I read `Foramt`. There's a typo :)

Comment (talregev) [inline cv_bridge/include/cv_bridge/cv_bridge.h]: O sorry, you are right :)

Comment (talregev) [inline cv_bridge/include/cv_bridge/cv_bridge.h]: @vrabaud  change and squash ;)

File: cv_bridge/include/cv_bridge/cv_bridge.h [hunk 2]
```diff
@@ -55,6 +55,16 @@ class CvImage;
...
 
+//from: http://docs.opencv.org/modules/highgui/doc/reading_and_writing_images_and_video.html#Mat imread(const string& filename, int flags)
+typedef enum {
+	BMP, DIP,
+	JPG, JPEG, JPE,
+	JP2,
+	PNG,
+	PBM, PGM, PPM,
+	SR, RAS,
+	TIFF, TIF,
```
Comment (vrabaud) [inline cv_bridge/include/cv_bridge/cv_bridge.h:66]: TIF should be before TIFF in alphabetical order

Comment (vrabaud) [inline cv_bridge/include/cv_bridge/cv_bridge.h:66]: does not matter actually, plus PNG already was before. That's fine no worries.

File: cv_bridge/src/cv_bridge.cpp [hunk 3]
```diff
@@ -409,14 +409,49 @@ cv::Mat matFromImage(const sensor_msgs::CompressedImage& source)
...
+		case RAS:
+			return "ras";
+		case SR:
+			return "sr";
+		case TIF:
+			return "tif";
+		case TIFF:
+			return "tiff";
+	}
+	return "";
```
Comment (vrabaud) [inline cv_bridge/src/cv_bridge.cpp]: should be throw an error here maybe ?

Comment (talregev) [inline cv_bridge/src/cv_bridge.cpp]: I don't think so, because it enum, I cover all the possibilities , user cannot enter a wrong format.

Comment (talregev) [inline cv_bridge/src/cv_bridge.cpp]: If you insist I will add a a throw there.
Any more feedback?

Comment (vrabaud) [inline cv_bridge/src/cv_bridge.cpp]: It's more that the behavior is undefined right ? What if I put an int that is bigger than the biggest enum in Format ? You can also have a warning message that says the format is not supported and default to JPG.

**Comment by talregev** [comment]
ok

File: cv_bridge/src/cv_bridge.cpp (hunk 3 above)
Comment (talregev) [inline cv_bridge/src/cv_bridge.cpp]: ok. add a throw.

Comment (talregev) [inline cv_bridge/src/cv_bridge.cpp]: How I write a warning message?
Write to std error?

**Comment by talregev** [comment]
thanks :)

**Pull request merged**